qiskit_ibm_runtime/options/resilience_options.py

```diff
@@ -85,26 +146,107 @@ def validate_resilience_options(resilience_options: dict) -> None:
+                raise ValueError(
+                    f"Unsupported value {extrapolator} for extrapolator. "
+                    f"Supported values are {get_args(ExtrapolatorType)}"
+                )
+
         if (
             extrapolator == "QuarticExtrapolator"
             and len(resilience_options.get("noise_factors")) < 5
         ):
             raise ValueError("QuarticExtrapolator requires at least 5 noise_factors.")
         if extrapolator == "CubicExtrapolator" and len(resilience_options.get("noise_factors")) < 4:
             raise ValueError("CubicExtrapolator requires at least 4 noise_factors.")
+
+        # Validation of new ZNE options
+        if resilience_options.get("zne_mitigation"):
+            # Validate extrapolator
+            extrapolator = resilience_options.get("zne_extrapolator")
+            if isinstance(extrapolator, str):
+                extrapolator = (extrapolator,)
+            if extrapolator is not None:
+                for extrap in extrapolator:
+                    if extrap not in get_args(ExtrapolatorType):
+                        raise ValueError(
+                            f"Unsupported value {extrapolator} for zne_extrapolator. "
+                            f"Supported values are {get_args(ExtrapolatorType)}"
+                        )
+
+            # Validation of noise factors
+            factors = resilience_options.get("zne_noise_factors")
+            if not isinstance(factors, (list, tuple)):
+                raise TypeError(
+                    f"zne_noise_factors option value must be a sequence, not {type(factors)}"
+                )
+            if any(i <= 0 for i in factors):
+                raise ValueError("zne_noise_factors` option value must all be non-negative")
+            if len(factors) < 1:
+                raise ValueError(f"zne_noise_factors cannot be empty")
+            if extrapolator is not None:
+                required_factors = {
+                    "exponential": 2,
+                    "double_exponential": 4,
+                    "linear": 2,
+                    "polynomial_degree_1": 2,
+                    "polynomial_degree_2": 3,
+                    "polynomial_degree_3": 4,
+                    "polynomial_degree_4": 5,
+                }
+                for extrap in extrapolator:
+                    if len(factors) < required_factors[extrap]:
+                        raise ValueError(
+                            f"{extrap} requires at least {required_factors[extrap]} zne_noise_factors"
+                        )
+
+            # Validation of threshold
+            threshold = resilience_options.get("zne_stderr_threshold")
+            if threshold is not None and threshold <= 0:
+                raise ValueError("Invalid zne_stderr_threshold option value must be > 0")
+
+        if resilience_options.get("pec_mitigation"):
+            if resilience_options.get("pec_mitigation"):
```

Comment:
One of these two options should be zne_mitigation?
```suggestion
        if resilience_options.get("pec_mitigation"):
            if resilience_options.get("zne_mitigation"):
```